met-web/src/components/engagement/form/EngagementWidgets/WhoIsListeningForm.tsx

```diff
@@ -8,15 +8,34 @@ import { postWidgetItems } from 'services/widgetService';
 import { WidgetDrawerContext } from './WidgetDrawerContext';
 import ContantInfoPaper from './ContactInfoPaper';
 import { WidgetType } from 'models/widget';
+import update from 'immutability-helper';
 
 const WhoIsListeningForm = () => {
-    const { handleWidgetDrawerOpen, handleAddContactDrawerOpen, loadingContacts, contacts, widgets } =
+    const { handleWidgetDrawerOpen, handleAddContactDrawerOpen, loadingContacts, contacts, widgets, clearSelected } =
         useContext(WidgetDrawerContext);
     const dispatch = useAppDispatch();
     const [selectedContact, setSelectedContact] = useState<Contact | null>(null);
     const [addedContacts, setAddedContacts] = useState<Contact[]>([]);
     const [addingWidgetItems, setAddingWidgetItems] = useState(false);
 
+    useEffect(() => {
+        const filteredContacts = contacts.filter((c: Contact) =>
```

Comment:
what's the logic here?

Also https://github.com/ryanmcdermott/clean-code-javascript#avoid-mental-mapping